src/msw/fontdlg.cpp

```diff
@@ -158,7 +170,24 @@ int wxFontDialog::ShowModal()
     if ( ChooseFont(&chooseFontStruct) != 0 )
     {
         wxRGBToColour(m_fontData.m_fontColour, chooseFontStruct.rgbColors);
-        m_fontData.m_chosenFont = wxFont(wxNativeFontInfo(logFont, this));
+
+        // Do the reverse transformation to the one done above here and
+        // reinterpret LOGFONT height using the standard DPI.
+        //
+        // It doesn't really matter what do we pass as "window" argument to
+        // wxNativeFontInfo ctor, as it's only used to initialize the point
+        // size, but we're going to overwrite it anyhow, but it's better to use
+        // NULL rather than "this" because it avoids a debug warning about not
+        // being able to determine the correct DPI for this window, due to not
+        // having any valid HWND for it.
+        wxNativeFontInfo nfi(logFont, NULL);
+        nfi.pointSize = wxNativeFontInfo::GetPointSizeAtPPI
+                        (
+                            logFont.lfHeight,
+                            wxDisplay::GetStdPPIValue()
+                        );
```

Comment:
This is wrong. The point size is already correct, it is `lf.lfHeight` that is wrong (based on 96dpi). You could use `WXAdjustToPPI` to fix it, using the dpi determined before calling `AutoSystemDpiAware`.

I'm not sure if it is needed. When the font is used in wxWidgets it is always adjusted to the DPI, but if a user uses the raw font data with `GetNativeFontInfo()` they might get the wrong `lfHeight`.

Reply:
Thanks, this is indeed wrong or, at least, useless, as doing this doesn't change `pointSize` value. I think it's better to use consistent values for `LOGFONT::lfHeight` too, just in case, so I've updated the PR to do it, thanks again for noticing this!